### ISSUE TITLE
db config generated by /manage/v3 rejected by mlDeploy/mlPreviewDeploy

### ISSUE THREAD

**Issue by kcoleman-marklogic** [marklogic-community/ml-gradle#465]
This is probably user head gap, rather than a bug, but since I'm not sure, maybe we can hash it out here. 

I'm creating a gradle project out of an existing ML configuration, so I used GET /manage/v3 to pull a config zip file. I then unpacked it, copied the conf files to my project, made a few changes and tried `mlPreviewDeploy`.

It failed with the following error:
```
> Unable to read resource payload: Cannot deserialize instance of `java.lang.String` out of START_OBJECT token
   at [Source: (String)"{
    "database-name": "apidoc-content",
    "security-database": "Security",
    "schema-database": "Schemas",
    "enabled": true,
  [snip]
    "fast-diacritic-sensitive-sea"[truncated 4447 chars]; line: 159, column: 5] (through reference chain: com.marklogic.mgmt.api.database.Database["forest"]->java.util.ArrayList[0])
```

I eventually figured out that the problem was that the db config files embedded the forest conf inside the db conf.  I didn't add that. It came out of /manage/v3 that way. That is, the db conf file included a "forest" property like this:

```
  "forest": [
    {
      "forest-name": "apidoc-content-1",
      "enabled": true,
      "host": "%%mlHost%%",
      "data-directory": "",
      "large-data-directory": "",
      "fast-data-directory": "",
      "fast-data-max-size": 0,
      "updates-allowed": "all",
      "availability": "online",
      "rebalancer-enable": true,
      "range": null,
      "failover-enable": true,
      "database-replication": null,
      "database": "apidoc-content"
    }
  ]
```

Here's where the user head gap probably comes in: I asked /manage/v3 to gen conf for both databases and forests. My request looked like this:

> http://localhost:8002/manage/v3?resource-name=RunDMC-\*\&format=zip\&resource-type=forest,database,server

Maybe that is why I ended up with the forest conf in the db conf files, as well as standalone forest conf files under config/forests/custom-forests (in the conf zip). They're certainly redundant, and maybe that's what gave ml-gradle heartburn. 

Should I have left off the forest resource in extracting the conf? Or not copied the standalone forest conf files to src/main/ml-config/forests/custom-forests? Or is it a problem with GET /manage/v3?

(The error reporting could have been more friendly if the redundancy was the source of trouble.)

ML 9.0-9, ml-gradle 3.14, Java 9, RHEL 7, gradle 5.4.1

**Comment by kcoleman-marklogic** [comment]
Maybe I should back up and ask how one is intended to use the zip generated by /manage/v3 with ml-gradle, since I think it's clearly not how I'm trying to use it.

I noticed that, though my deployment was succeeding after I made the change above, it was generating an extra forest for each db. The one I configured in `src/main/ml-config/forests/custom-forests/` (from `config/forests/custom-forests in the generated zip`), and some default forest. 

I ended up removing the custom forest config files to stop this.
